### ISSUE TITLE
implement AnalogSignalArray.describe()

### ISSUE THREAD

**Issue by eackermann** [eackermann/nelpy#144]
Similar to pandas, but segment-aware.

**Comment by eackermann** [comment]
One possibility is to append the columns with EpochArray descriptive summaries?

**Comment by shayokdutta** [comment]
I agree. Having a short description per signal in ASA plus descriptions of each epoch is probably ideal. Obviously, all of these should be optional and a singular epoch description should be an option.

**Comment by eackermann** [comment]
No I think you misunderstood, although I have also requested what you describe as a separate feature.

This issue is about replicating the pandas.describe() functionality which is analogous to the dataframe.summary() in R

**Comment by eackermann** [comment]
Mhmm. I only saw your reply and the issue title in an email, so I forgot I already mentioned that it should be pandas-like. So I guess you understood correctly.

Proposed interface could be

```python
asa.describe() # overview with n_signals, n_samples, n_epochs, duration, signal min, signal max, signal mean, hasnan, within-epoch (-min, mean, max)

asa.describe(all=True) # describe most of the above, but per epoch, across all signals

asa._subset(...).describe(all=True) # describe for a particular subset of signals, or most likely, for a single signal. This interface needs more work / thought :/
```
